hudi-common/src/main/java/org/apache/hudi/common/util/CommitUtils.java

```diff
@@ -66,7 +66,8 @@ public static HoodieCommitMetadata buildMetadata(List<HoodieWriteStat> writeStat
     if (extraMetadata.isPresent()) {
       extraMetadata.get().forEach(commitMetadata::addMetadata);
     }
-    commitMetadata.addMetadata(HoodieCommitMetadata.SCHEMA_KEY, schemaToStoreInCommit);
+    // do we need to set empty string for schema of schema is null ?
+    commitMetadata.addMetadata(HoodieCommitMetadata.SCHEMA_KEY, schemaToStoreInCommit == null ? "" : schemaToStoreInCommit);
```

Comment:
do we need to set empty string for schema if schema is null ? Better to fix at the datasource layer to set the schema properly rather than silently setting empty string here.